homeassistant/components/zwave_mqtt/services.py

```diff
@@ -0,0 +1,173 @@
+"""Methods and classes related to executing Z-Wave commands and publishing these to hass."""
+import logging
+
+from openzwavemqtt.const import CommandClass, ValueType
+import voluptuous as vol
+
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from . import const
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ZWaveServices:
+    """Class that holds our services ( Zwave Commands) that should be published to hass."""
+
+    def __init__(self, hass, manager, data_nodes):
+        """Initialize with both hass and ozwmanager objects."""
+        self._hass = hass
+        self._manager = manager
+        self._data_nodes = data_nodes
+
+    @callback
+    def register(self):
+        """Register all our services."""
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_ADD_NODE,
+            self.add_node,
+            schema=vol.Schema(
+                {
+                    vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+                    vol.Optional(const.ATTR_SECURE, default=False): vol.Coerce(bool),
```

Comment:
I don't know enough about Z-Wave, but secure defaulting to true sounds more secure?

Reply:
Not really, secure in OZW is not always good. Use it with care.

Reply:
Secure in Z-Wave is only really important for locks and covers, it adds a lot of communication overhead and can cause issues if there are too many secure devices since Z-Wave is limited on bandwidth

Reply:
Let's keep the default as it is.